cypress/e2e/article.cy.js

```diff
@@ -1,24 +1,184 @@
+  });
+
+  it('The user can create a new article with filled all field', () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder('Article Title').type(articleTitle);
+    cy.findByPlaceholder(`What's this article about?`).type(articleDescription);
+    cy.findByPlaceholder(`Write your article (in markdown)`).type(articleBody);
+    cy.findByPlaceholder(`Enter tags`).type(`${articleTags}{enter}`);
+    cy.contains('button', 'Publish Article').click();
+    cy.wait(2000);
+    cy.get('h1').should('contain.text', articleTitle);
+    cy.get('p').should('contain.text', articleBody);
+  });
+
+  it('The user can create new article using only title', () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder('Article Title').type(articleTitle);
+    cy.contains('button', 'Publish Article').click();
+    cy.wait(2000);
+    cy.get('h1').should('contain.text', articleTitle);
+  });
+
+  it('The user can create new article using title and description', () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder('Article Title').type(articleTitle);
+    cy.findByPlaceholder(`What's this article about?`).type(articleDescription);
+    cy.contains('button', 'Publish Article').click();
+  });
+
+  it('The user can create new article using all field without Tags ', () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder('Article Title').type(articleTitle);
+    cy.findByPlaceholder(`What's this article about?`).type(articleDescription);
+    cy.findByPlaceholder(`Write your article (in markdown)`).type(articleBody);
+    cy.contains('button', 'Publish Article').click();
+  });
+
+  it(`The user can't create new article without title`, () => {
+    const { articleTitle, articleDescription, articleBody, articleTags }
+      = generateNewArticle();
+
+    cy.findByPlaceholder(`What's this article about?`).type(articleDescription);
+    cy.findByPlaceholder(`Write your article (in markdown)`).type(articleBody);
+    cy.contains('button', 'Publish Article').click();
+    cy.contains('.swal-modal', 'Oops!').should('exist');
   });
+});
 
+describe(`Edit article`, () => {
   beforeEach(() => {
     cy.task('db:clear');
+    cy.registerNewUser().then(user => {
+      cy.login(user).then(() => {
+        cy.createNewArticle(user).then((slug) => {
+          cy.visit(`#/articles/${slug}`);
+          user.articleSlug = slug;
+        });
+      });
+    }).as('user');
   });
 
-  it('should be created using New Article form', () => {
+  it('The user can edit whole of their own article.', () => {
+    cy.get('@user').then(user => {
+      const newTitle = faker.lorem.sentence(2).slice(0, -1);
```

Comment:
It can be moved to the commands because you reuse this code in the following tests

Reply:
Check everywhere